### PR TITLE
Add Summary Count for History Endpoints

### PR DESCRIPTION
**Pull request by mikaelweave**
## Description
Adds `?_summary=count` and `?_count=0` to `/_history` endpoints to enable counting all versions (historical and soft deleted) of resources.

## Related issues
AB113823

## Testing
Tested locally with historical and deleted resources

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- Tag the PR with **Schema Version backward compatible** or **Schema Version backward incompatible** or **Schema Version unchanged** if this adds or updates Sql script which is/is not backward compatible with the code.
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

**Comment by SergeyGaluzo** [comment]
Do we have E2E test that call count on history end point?

**Comment by mikaelweave** [comment]
@SergeyGaluzo - I can add if you think it's useful! I was hesitating adding more E2E tests for repo bloat (shift left testing) but can add quickly. 😄

**Comment by SergeyGaluzo** [comment]
I would combine with some existing test that creates appropriate data, and check 2 things...

---
In reply to: [1884005353](https://github.com/microsoft/fhir-server/pull/3661#issuecomment-1884005353) [](http://example.com/codeflow?ancestors=1884005353)

**Review by SergeyGaluzo — APPROVED** [review]
:shipit:

**Review by SergeyGaluzo — DISMISSED** [review]

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Export/ExportDataTests.cs [hunk 1]
```diff
@@ -188,6 +188,28 @@ public async Task GivenFhirServer_WhenDataIsExportedWithHistoryAndSoftDeletesNot
             await ExportAndSoftDeleteTestHelper(parallel: false, history: true, deletes: true);
         }
 
+        // No need to test both code paths for testing container is written to.
+        [Fact]
+        public async Task GivenFhirServer_WhenSummarizingHistoricalData_ThenCountIsReturnedWethHistory()
```
Comment (LTA-Thinking) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Export/ExportDataTests.cs:193]: Nit: typo Weth

File: src/Microsoft.Health.Fhir.Core/Features/Search/SearchService.cs [hunk 2]
```diff
@@ -157,6 +159,13 @@ protected SearchService(ISearchOptionsFactory searchOptionsFactory, IFhirDataSto
             {
                 queryParameters.Add(Tuple.Create(KnownQueryParameterNames.Count, count.ToString()));
             }
+            else
+            {
+                if ((count.HasValue && count == 0) || (summary is not null && summary.Equals(SummaryType.Count.ToString(), StringComparison.OrdinalIgnoreCase)))
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.Core/Features/Search/SearchService.cs:164]: Nit: why not an else if?

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Export/ExportDataTests.cs (hunk 1 above)
Comment (LTA-Thinking) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Export/ExportDataTests.cs:193]: Why is this in the export tests? This looks like it should be in HistoryTests.

Comment (mikaelweave) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Export/ExportDataTests.cs]: This was a suggestion above - I was concerned about E2E performance but can rewrite the test to go off less data. I haven't looked at history tests but will check it out.

**Review by LTA-Thinking — DISMISSED** [review]

**Review by LTA-Thinking — DISMISSED** [review]

**Review by LTA-Thinking — APPROVED** [review]

**Pull request merged**